### ISSUE TITLE
cache sem is not taken before subCacheItemLookup() call

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#2882]
In `mongoUpdateSubscription()` function we have:

```
  if (!noCache)
  {
    subCacheP = subCacheItemLookup(tenant.c_str(), subUp.id.c_str());
  }
```

This access to cache is not protected by sem taken / give. Potentially, it may cause a segfault.

**Comment by fgalan** [comment]
Flow diagram MB-12 should be changed once this issue gets completed.

**Comment by SourabhChourasiya-NEC** [comment]
Fixed in #3373

**Comment by fgalan** [comment]
Doc fixes in PR #3376
